controller/controller.go

```diff
@@ -207,17 +207,52 @@ func (c *controller) updateIngresses() error {
 	return nil
 }
 
-type serviceName struct {
+func validate(e *IngressEntry) error {
+	if e.Host == "" {
+		return errors.New("missing host")
+	}
+	if len(e.Service.Addresses) == 0 {
+		return errors.New("no service endpoints")
+	}
+	if e.Service.Port == 0 {
+		return errors.New("missing service port")
+	}
+	return nil
+}
+
+type serviceKey struct {
 	namespace string
 	name      string
+	port      int32
 }
 
-func mapNamesToAddresses(services []*v1.Service) map[serviceName]string {
-	m := make(map[serviceName]string)
+func mapEndpointsToServices(multipleEndpoints []*v1.Endpoints) map[serviceKey]Service {
+	m := make(map[serviceKey]Service)
+
+	for _, endpoints := range multipleEndpoints {
+		for _, subset := range endpoints.Subsets {
+			for _, port := range subset.Ports {
+
+				if port.Protocol == v1.ProtocolUDP {
```

Comment:
Yep protocol is optional and can be blank. So we just want the explicit check here.